src/devtools/client/debugger/src/components/SecondaryPanes/Frames/NewFrames.tsx

```diff
@@ -31,38 +30,16 @@ function FramesRenderer({
   panel,
   pauseId,
   asyncIndex = 0,
-  copyStackTrace,
+  getFullStackTrace,
 }: {
-  panel: "debugger" | "console" | "webconsole";
+  panel: CommonFrameComponentProps["panel"];
   pauseId: PauseId;
   asyncIndex?: number;
-  copyStackTrace: () => void;
+  getFullStackTrace: () => PauseFrame[];
 }) {
-  const cx = useAppSelector(getThreadContext);
+  const commonProps = useGetCommonFrameComponentProps(panel, getFullStackTrace);
```

Comment:
Why do these props need to be fetched outside and passed into `PauseFrames` ? I think the API would be a bit simpler if `PauseFrames` required a `panel: CommonFrameComponentProps["panel"]` prop and fetched them itself.